app/vmselect/prometheus/prometheus.go

```diff
@@ -1133,3 +1142,27 @@ func getCommonParams(r *http.Request, startTime time.Time, requireNonEmptyMatch
 	}
 	return cp, nil
 }
+
+func errWithContext(err error, context string) error {
+	if err == nil {
+		return nil
+	}
+	str := err.Error()
```

Comment:
This loses the information about wrapped errors in `err`. Probably it would be better using `errors.Unwrap` in order to preserve the structure of `err` (if this is possible)